services/spar/test-integration/Test/Spar/APISpec.hs

```diff
@@ -638,21 +638,97 @@ specCRUDIdentityProvider = do
           callIdpUpdate' (env ^. teSpar) (Just owner) idpid (IdPMetadataValue "<NotSAML>bloo</NotSAML>" undefined)
             `shouldRespondWith` ((== 400) . statusCode)
-    describe "issuer changed to one that already exists in another team" $ do
+    describe "issuer changed to one that already exists in *another* team" $ do
       it "rejects" $ do
         env <- ask
         (owner1, _, (^. idpId) -> idpid1) <- registerTestIdP
         (_, _, _, (IdPMetadataValue _ idpmeta2, _)) <- registerTestIdPWithMeta
         callIdpUpdate' (env ^. teSpar) (Just owner1) idpid1 (IdPMetadataValue (cs $ SAML.encode idpmeta2) undefined)
-          `shouldRespondWith` checkErr (== 400) "idp-used-in-other-team"
-    describe "issuer changed to one that is new" $ do
+          `shouldRespondWith` checkErr (== 400) "idp-issuer-in-use"
+    describe "issuer changed to one that already exists in *the same* team" $ do
       it "rejects" $ do
         env <- ask
-        (owner, _, (^. idpId) -> idpid, (IdPMetadataValue _ idpmeta, _)) <- registerTestIdPWithMeta
-        newissuer <- makeIssuer
-        let idpmeta' = idpmeta & edIssuer .~ newissuer
-        callIdpUpdate' (env ^. teSpar) (Just owner) idpid (IdPMetadataValue (cs $ SAML.encode idpmeta') undefined)
-          `shouldRespondWith` checkErr (== 400) "cannot-update-idp-issuer"
+        (owner1, _, (^. idpId) -> idpid1, (IdPMetadataValue _ idpmeta1, _)) <- registerTestIdPWithMeta
+        (idpmeta2, _) <- makeTestIdPMetadata
+        _ <- call $ callIdpCreate (env ^. teSpar) (Just owner1) idpmeta2
+        let idpmeta3 = idpmeta1 & edIssuer .~ (idpmeta2 ^. edIssuer)
+        callIdpUpdate' (env ^. teSpar) (Just owner1) idpid1 (IdPMetadataValue (cs $ SAML.encode idpmeta3) undefined)
+          `shouldRespondWith` checkErr (== 400) "idp-issuer-in-use"
+    describe "issuer changed to one that is new" $ do
+      let tryLogin :: SignPrivCreds -> IdP -> TestSpar SAML.UserRef
+          tryLogin privkey idp = do
+            userSubject <- SAML.unspecifiedNameID . UUID.toText <$> liftIO UUID.nextRandom
+            tryLoginWith privkey idp userSubject
+          tryLoginWith :: SignPrivCreds -> IdP -> NameID -> TestSpar SAML.UserRef
+          tryLoginWith privkey idp userSubject = do
+            env <- ask
+            spmeta <- getTestSPMetadata
+            (_, authnreq) <- call $ callAuthnReq (env ^. teSpar) (idp ^. SAML.idpId)
+            idpresp <- runSimpleSP $ mkAuthnResponseWithSubj userSubject privkey idp spmeta authnreq True
+            sparresp <- submitAuthnResponse idpresp
+            liftIO $ do
+              statusCode sparresp `shouldBe` 200
+              let bdy = maybe "" (cs @LBS @String) (responseBody sparresp)
+              bdy `shouldContain` "<title>wire:sso:success</title>"
+            either (error . show) (pure . view userRefL) $
+              SAML.parseFromDocument (fromSignedAuthnResponse idpresp)
+      context "impure (overwrite old idp)" $ do
```

Comment:
can we call this "update existing idp"  instead of impure (overwrite old idp).  I think this is easier to understand for the reader.  to describe what we're testing in terms of the rest 'semantics'

Reply:
in current form it makes the reader ask questions,  (why is it impure? why is creation more pure than updating? etc)

Reply:
hum, i just realized that this `context` makes no sense any more, so i removed it...